velox/exec/Task.h

```diff
@@ -641,6 +641,16 @@ class Task : public std::enable_shared_from_this<Task> {
   /// node IDs specified in split management methods.
   const std::unordered_set<core::PlanNodeId> splitPlanNodeIds_;
 
+  std::unique_ptr<memory::MemoryPool> pool_;
```

Comment:
It might be cleaner to move exchangClientXxx member variable. 

Consider adding a comment just above pool_ saying that all member variables that hold references to pool_ must be defined after pool_.